worker/uniter/resolver.go

```diff
@@ -69,14 +68,6 @@ func (s *uniterResolver) NextOp(
 		if localState.Conflicted {
 			return s.nextOpConflicted(localState, remoteState, opFactory)
 		}
-		op, err = s.config.UpgradeCharmProfile.NextOp(localState, remoteState, opFactory)
```

Comment:
This is basically fine right, because you can never have a controller behind an agent. So new controllers would never emit the change event, so new agents never need this code as it'll be dead code anyway. That's my understanding of this PR?

Reply:
yes.  we only need to keep the watcher in state and compatibility in the apiserver layer for 2.5.x uniters to work with 2.6 controllers until those models can be upgraded.